src/controls/editor-mode/components/color-scheme-controls.js

```diff
@@ -118,32 +123,45 @@ const ColorSchemeControls = () => {
   // common state
   const presetColors = useStoreState((state) => state.presetColors)
   const baseColor = useStoreState((state) => state.genericOptions.baseColor)
+  const widgetBaseColor1Selection = useStoreState((state) => state.ui.widgetBaseColor1Selection)
   const colorRepresentation = useStoreState((state) => state.ui.colorRepresentation)
   const type = useStoreState((state) => state.type)
 
+  const [color, selectedColor] = useMemo(() => widgetBaseColor1Selection
+    ? [baseColor.color1, Object.keys(baseColor)[0]]
+    : [baseColor.color2, Object.keys(baseColor)[1]]
+  ,[widgetBaseColor1Selection, baseColor])
+
   // local state
   const [selectedColorIndex, setSelectedColorIndex] = useState(
-    presetColors.indexOf(baseColor) === -1
+    presetColors.indexOf(color) === -1
       ? presetColors.length - 1
-      : presetColors.indexOf(baseColor)
+      : presetColors.indexOf(color)
   )
   const [inputError, setInputError] = useState(false)
   const [showPicker, setShowPicker] = useState(false)
   const [showInputHelper, setShowInputHelper] = useState(false)
 
-  const styles = useStyles({ baseColor, showPicker, type })
+  const styles = useStyles({ color, showPicker, type })
 
   const updateBaseColor = useDebouncedCallback(v => userUpdate({
-    genericOptions: { baseColor: colord(v).toHex() },
+    genericOptions: { baseColor: { [selectedColor]: colord(v).toHex() } },
   }), 100)
 
   useEffect(() => {
-    update({ presetColors: presetColors.map((_c, i) => i === selectedColorIndex ? baseColor : _c) })
+    update({ presetColors: presetColors.map((_c, i) => i === selectedColorIndex ? color : _c) })
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [baseColor, selectedColorIndex, update])
+  }, [color, selectedColorIndex, update])
```

Comment:
The lint suppression comment before this one isn't needed anymore it seems

Reply:
![Screen Shot 2022-12-20 at 12 54 06 PM](https://user-images.githubusercontent.com/41120953/208768550-8d6ea5d7-f62d-44d4-b09e-cffe11574ba1.png)
@woozyking it's still needed. For some reason Kyle decided not to include `presetColors` in the dep list. I will look at this later and keep it as it is.

Reply:
ah I see, fair enough